jerry-core/ecma/base/ecma-alloc.c

```diff
@@ -128,7 +128,7 @@ ecma_dealloc_extended_object (ecma_object_t *object_p, /**< extended object */
  *
  * @return pointer to allocated memory
  */
-inline ecma_string_t * JERRY_ATTR_ALWAYS_INLINE
+extern JERRY_ALWAYS_INLINE ecma_string_t *
```

Comment:
Could you explain me why the `extern` keyword is always required in C files as well?

Reply:
> Could you explain me why the `extern` keyword is always required in C files as well?

Because in msvc, inline without extern would  cause the function don't generate symbol and then cause linkage error.